app/models/miq_server/role_management.rb

```diff
@@ -50,7 +50,7 @@ def set_active_role_flags
   end
 
   def set_assigned_roles
-    self.role = VMDB::Config.new("vmdb").config[:server][:role]
+    self.role = ::Settings.server.role
```

Comment:
@Fryguy as we have checked, there are 3 places calling ```set_assigned_roles```:
```MiqServer#start```       - no race
```MiqSerber.atStart```    - no race
```MiqServer#monitor``` - called after config was changed and Settings was reloaded

looks like we are save here